cql3/expr/to_restriction.cc

```diff
@@ -123,6 +125,156 @@ ::shared_ptr<restrictions::restriction> new_subscripted_column_restriction(subsc
+                                                                            std::move(prepared_rhs),
+                                                                            order);
+    }
+
+    if (oper == oper_t::IN) {
+        if (auto rhs_marker = as_if<bind_variable>(&prepared_rhs)) {
+            return ::make_shared<restrictions::multi_column_restriction::IN_with_marker>(schema,
+                                                                                         std::move(lhs_cols),
+                                                                                         std::move(*rhs_marker));
+        }
+
+        if (auto rhs_list = as_if<collection_constructor>(&prepared_rhs)) {
+            return ::make_shared<restrictions::multi_column_restriction::IN_with_values>(schema,
+                                                                                         std::move(lhs_cols),
+                                                                                         std::move(rhs_list->elements));
+        }
+
+        if (auto rhs_list = as_if<constant>(&prepared_rhs)) {
+            const list_type_impl* list_type = dynamic_cast<const list_type_impl*>(&rhs_list->type->without_reversed());
+            const data_type& elements_type = list_type->get_elements_type();
+
+            utils::chunked_vector<managed_bytes> raw_elems = get_list_elements(*rhs_list);
+            std::vector<expression> list_elems;
+            list_elems.reserve(raw_elems.size());
+
+            for (managed_bytes elem : std::move(raw_elems)) {
+                raw_value elem_val = raw_value::make_value(std::move(elem));
+                list_elems.emplace_back(constant(elem_val, elements_type));
+            }
+
+            return ::make_shared<restrictions::multi_column_restriction::IN_with_values>(schema,
+                                                                                         std::move(lhs_cols),
+                                                                                         std::move(list_elems));
+        }
+    }
+
+    on_internal_error(expr_logger,
+                      fmt::format("new_multi_column_restriction operation type: {} not handled", oper));
+}
+
+void preliminary_binop_vaidation_checks(const binary_operator& binop) {
+    if (auto lhs_tup = as_if<tuple_constructor>(&binop.lhs)) {
+        if (binop.op == oper_t::CONTAINS) {
+            throw exceptions::invalid_request_exception("CONTAINS cannot be used for Multi-column relations");
+        }
+
+        if (binop.op == oper_t::CONTAINS_KEY) {
+            throw exceptions::invalid_request_exception("CONTAINS_KEY cannot be used for Multi-column relations");
+        }
+
+        if (binop.op == oper_t::LIKE) {
+            throw exceptions::invalid_request_exception("LIKE cannot be used for Multi-column relations");
+        }
+
+        if(auto rhs_tup = as_if<tuple_constructor>(&binop.rhs)) {
```

Comment:
`if` is not a function.

Reply:
I don't understand what's wrong here

Reply:
function: `f(...)`
not-function: `if (...)`

Reply:
You mean that there should be a space after if?
I added a space

Reply:
yes